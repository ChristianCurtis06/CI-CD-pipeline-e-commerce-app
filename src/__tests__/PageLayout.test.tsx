import PageLayout from "../components/PageLayout";
import { render } from "@testing-library/react";

jest.mock('../components/NavBar', () => () => <div>Mock NavBar</div>);

describe('PageLayout Component', () => {
    test('renders children and matches the snapshot', () => {
        const { asFragment, getByText } = render(
            <PageLayout>
                <div>Mock Child Content</div>
            </PageLayout>
        );

        expect(getByText("Mock Child Content")).toBeInTheDocument();
        expect(getByText("Mock NavBar")).toBeInTheDocument();

        expect(asFragment()).toMatchSnapshot();
    });
});