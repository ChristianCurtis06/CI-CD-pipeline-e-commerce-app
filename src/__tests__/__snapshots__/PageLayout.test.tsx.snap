// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout Component renders children and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-0 container-fluid"
  >
    <div
      class="col"
    >
      <div>
        Mock NavBar
      </div>
    </div>
    <div>
      Mock Child Content
    </div>
  </div>
</DocumentFragment>
`;
